Sources/Fluent/Query/Comparison.swift

```diff
@@ -28,8 +28,20 @@ public func ==(_ lhs: Filter.Comparison, _ rhs: Filter.Comparison) -> Bool
     switch (lhs, rhs) {
     case (.like(let leftPos), .like(let rightPos)):
         return leftPos == rightPos
+    case (.equals, .equals):
```

Comment:
@tannernelson On the diff before, that failed, the reason was the comparison lhs == rhs was infinitely calling the override of the == operator. In order to overcome this infinite recursion I had to add a a case for each enum matching the other. Is there a better, short-hand way of doing this? Because, as of now, if there's ever another Comparison operator added and the author doesn't know to modify this overload and add their case that the catchall default: is going to catch their enum and the comparison is always going to be false. There has to be a better solution, I'm almost certain this is going to trip somebody in the future.